src/gui/wallet_test.go

```diff
@@ -0,0 +1,330 @@
+			"0",
+			0,
+			"",
+			nil,
+			nil,
+			wallet.BalancePair{},
+			nil,
+			nil,
+		},
+		{
+			"400 - bad dst addr",
+			"POST",
+			"/wallet/spend",
+			&httpBody{
+				Id:  "123",
+				Dst: "2konv5no3DZvSMxf2GPVtAfZinfwqCGhfVQ======",
+			},
+			http.StatusBadRequest,
+			"400 Bad Request - invalid destination address: Invalid address length",
+			"0",
+			0,
+			"",
+			nil,
+			nil,
+			wallet.BalancePair{},
+			nil,
+			nil,
+		},
+		{
+			"400 - no coins",
+			"POST",
+			"/wallet/spend",
+			&httpBody{
+				Id:  "123",
+				Dst: "2konv5no3DZvSMxf2GPVtAfZinfwqCGhfVQ",
+			},
+			http.StatusBadRequest,
+			"400 Bad Request - invalid \"coins\" value",
+			"0",
+			0,
+			"",
+			nil,
+			nil,
+			wallet.BalancePair{},
+			nil,
+			nil,
+		},
+		{
+			"400 - coins is string",
+			"POST",
+			"/wallet/spend",
+			&httpBody{
+				Id:    "123",
+				Dst:   "2konv5no3DZvSMxf2GPVtAfZinfwqCGhfVQ",
+				Coins: "foo",
+			},
+			http.StatusBadRequest,
+			"400 Bad Request - invalid \"coins\" value",
+			"0",
+			0,
+			"",
+			nil,
+			nil,
+			wallet.BalancePair{},
+			nil,
+			nil,
+		},
+		{
+			"400 - coins is negative value",
+			"POST",
+			"/wallet/spend",
+			&httpBody{
+				Id:    "123",
+				Dst:   "2konv5no3DZvSMxf2GPVtAfZinfwqCGhfVQ",
+				Coins: "-123",
+			},
+			http.StatusBadRequest,
+			"400 Bad Request - invalid \"coins\" value",
+			"0",
+			0,
+			"",
+			nil,
+			nil,
+			wallet.BalancePair{},
+			nil,
+			nil,
+		},
+		{
+			"400 - zero coins",
+			"POST",
+			"/wallet/spend",
+			&httpBody{
+				Id:    "123",
+				Dst:   "2konv5no3DZvSMxf2GPVtAfZinfwqCGhfVQ",
+				Coins: "0",
+			},
+			http.StatusBadRequest,
+			"400 Bad Request - invalid \"coins\" value, must > 0",
+			"0",
+			0,
+			"",
+			nil,
+			nil,
+			wallet.BalancePair{},
+			nil,
+			nil,
+		},
+		{
+			"200 - gw spend error",
+			"POST",
+			"/wallet/spend",
+			&httpBody{
+				Id:    "123",
+				Dst:   "2konv5no3DZvSMxf2GPVtAfZinfwqCGhfVQ",
+				Coins: "12",
+			},
+			http.StatusOK,
+			"400 Bad Request - invalid \"coins\" value, must > 0",
+			"123",
+			12,
+			"2konv5no3DZvSMxf2GPVtAfZinfwqCGhfVQ",
+			&coin.Transaction{},
+			errors.New("Spend error"),
+			wallet.BalancePair{},
+			nil,
+			&SpendResult{
+				Error: "Spend error",
+			},
+		},
+		{
+			"200 - gw GetWalletBalance error",
+			"POST",
+			"/wallet/spend",
+			&httpBody{
+				Id:    "1234",
+				Dst:   "2konv5no3DZvSMxf2GPVtAfZinfwqCGhfVQ",
+				Coins: "12",
+			},
+			http.StatusOK,
+			"400 Bad Request - Get wallet balance failed: GetWalletBalance error",
+			"1234",
+			12,
+			"2konv5no3DZvSMxf2GPVtAfZinfwqCGhfVQ",
+			&coin.Transaction{},
+			errors.New("GetWalletBalance error"),
+			wallet.BalancePair{},
+			nil,
+			&SpendResult{
+				Error: "GetWalletBalance error",
+			},
+		},
+		{
+			"200 - OK",
+			"POST",
+			"/wallet/spend",
+			&httpBody{
+				Id:    "1234",
+				Dst:   "2konv5no3DZvSMxf2GPVtAfZinfwqCGhfVQ",
+				Coins: "12",
+			},
+			http.StatusOK,
```

Comment:
should not be `StatusOK`

Reply:
Then we need make some changes [here](https://github.com/skycoin/skycoin/blob/6634757f5dbf49be580e8ad7a0862d25bcd2dedd/src/gui/wallet.go#L141), because here we receive SpendResult struct with Error field. But the status is 200 OK.